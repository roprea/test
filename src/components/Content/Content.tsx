import { lazy, Suspense } from 'react';
import type { HTMLAttributes} from 'react';
import { Navigate, Route, Routes } from 'react-router-dom';

const Dashboard = lazy(() => import('components/pages/Dashboard'));

function Content(props: Props) {
  return (
    <main {...props}>
      <Suspense
        fallback={
          <div>Loader</div>
        }
      >
        <Routes>
          <Route path="/" element={<Dashboard />} />
          <Route path="*" element={<Navigate to="/" replace />} />
        </Routes>
      </Suspense>
    </main>
  );
}

type Props = HTMLAttributes<HTMLElement>;

export default Content;
