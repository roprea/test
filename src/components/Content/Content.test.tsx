import { basicTestRunner } from 'test-utils/basic-test-runner';
import { act, renderHook, render, screen, waitFor } from 'test-utils/testing-library';
import { BrowserRouter, useNavigate } from 'react-router-dom';
import Content from './Content';

jest.mock('components/pages/Dashboard', () => ({
  __esModule: true,
  default: () => <div>Dashboard Page</div>,
}));

describe('Content', () => {
  basicTestRunner(
    <Content />,
    () => ({ waitForReady: () => () => screen.getByText('Dashboard Page') }),
    {
      wrapper: BrowserRouter,
    }
  );

  it('render dashboard page for root path', async () => {
    const { result } = renderHook(() => useNavigate(), { wrapper: BrowserRouter });
    act(() => result.current('/'));
    render(<Content />, { wrapper: BrowserRouter });
    await waitFor(() => {
      expect(screen.getByText('Dashboard Page')).toBeInTheDocument();
    });
  });
});
