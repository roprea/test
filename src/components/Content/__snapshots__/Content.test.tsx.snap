// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Content Basic component behaviour should match snapshot 1`] = `
<div>
  <main>
    <div>
      Dashboard Page
    </div>
  </main>
</div>
`;
